component/registry/registry.go

```diff
@@ -0,0 +1,63 @@
+package registry
```

Comment:
I believe we generally use the term `store` for things we're storing on disk. I don't know if that fits any better here.

I think it's kind of unfortunate that we can't re-use the term registry. This [really is a registry](https://en.wikipedia.org/wiki/Service_locator_pattern) and calling it anything else seems confusing.